### PR TITLE
Add v1 to v2 migration guide and improve README

### LINKED ISSUE #175: Add v1 to v2 migration guide

### PR DESCRIPTION
**Pull request by gtupak**
Closes #175

- Added v1 to v2 migration guide
- Removed references to `trackReferrer` and `trackUTM` from `README.md`
- Small refactors inside `ArcxAnalyticsSdk.ts`. Nothing worth making a release for.
- Update the `example/arcx-provider` app to use sdk v2

### CODE REVIEW

File: example/cra-provider/src/Web3Buttons.tsx [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
+        // Disconnect
+        sdk.wallet({
+          chainId,
+          account: '',
+        })
+      }
+      knownAccount.current = account
+    } else if (knownChainId.current !== chainId && chainId) {
+      sdk.chain({ chainId })
+    }
```
Comment (classicus-eth) [inline example/cra-provider/src/Web3Buttons.tsx:33]: Should this be an `else`? i.e. is it possible to have a new account AND and new chain in a single useEffect() refresh? 
```suggestion
    } 
    
    if (knownChainId.current !== chainId && chainId) {
      sdk.chain({ chainId })
    }
```

Comment (classicus-eth) [reply example/cra-provider/src/Web3Buttons.tsx:33]: Just noticed this is in example code, not really that big a deal ^^ 😅

**Review by classicus-eth — APPROVED** [review]
LGTM! Quick unrelated-question in example code, feel free to ignore

File: example/cra-provider/src/Web3Buttons.tsx (hunk 1 above)
Comment (gtupak) [reply example/cra-provider/src/Web3Buttons.tsx:33]: > is it possible to have a new account AND and new chain in a single useEffect() refresh?
No it shouldn't but I tested out this code and it worked well enough, so I'd say it's good enough for the example :)

Comment (gtupak) [reply example/cra-provider/src/Web3Buttons.tsx:33]: If we do it this way, there will be a CONNECT followed by a CHAIN_CHANGED immediately, so I would leave it as is, esp given this is an example repo

**Pull request merged**